iceoryx2-ffi/cxx/tests/src/service_publish_subscribe_tests.cpp

```diff
@@ -943,9 +962,82 @@ TYPED_TEST(ServicePublishSubscribeTest,
     const auto service_name = iox2_testing::generate_service_name();
 
     auto node = NodeBuilder().create<SERVICE_TYPE>().expect("");
-    auto sut_create = node.service_builder(service_name).template publish_subscribe<TransmissionData>().create();
+    auto sut_create = node.service_builder(service_name).template publish_subscribe<Payload>().create();
 
-    auto sut_open = node.service_builder(service_name).template publish_subscribe<other::TransmissionData>().open();
+    auto sut_open = node.service_builder(service_name).template publish_subscribe<other::Payload>().open();
+    ASSERT_TRUE(sut_open.has_error());
+}
+
+TYPED_TEST(ServicePublishSubscribeTest, opening_existing_service_with_set_user_header_type_name_works) {
+    constexpr ServiceType SERVICE_TYPE = TestFixture::TYPE;
+    const auto service_name = iox2_testing::generate_service_name();
+
+    auto node = NodeBuilder().create<SERVICE_TYPE>().expect("");
+    auto sut_create = node.service_builder(service_name)
+                          .template publish_subscribe<uint8_t>()
+                          .template user_header<CustomHeader>()
+                          .create();
+    auto sut_open = node.service_builder(service_name)
+                        .template publish_subscribe<uint8_t>()
+                        .template user_header<CustomHeader>()
+                        .open();
+    ASSERT_FALSE(sut_open.has_error());
+}
+
+TYPED_TEST(ServicePublishSubscribeTest,
+           opening_existing_service_with_different_header_but_same_set_user_header_type_name_works) {
```

Comment:
Oh, could you test if the alignment and size check also prevents a misconnection when the type name is the same but the type size or type alignment is different.

* opening_existing_service_with_same_type_name_but_different_{size|alignment}_fails

And can you add such a test also for the payload type